nomad/structs/config/vault.go

```diff
@@ -0,0 +1,131 @@
+package config
+
+import vault "github.com/hashicorp/vault/api"
+
+// VaultConfig contains the configuration information necessary to
+// communicate with Vault in order to:
+//
+// - Renew Vault tokens/leases.
+//
+// - Pass a token for the Nomad Server to derive sub-tokens.
+//
+// - Create child tokens with policy subsets of the Server's token.
+type VaultConfig struct {
+
+	// RoleName is the Vault role in which Nomad will derive child tokens using
+	// /auth/token/create/[role_name]
+	RoleName string `mapstructure:"role_name"`
+
+	// RoleToken is the periodic Vault token given to Nomad such that it can
+	// derive child tokens. The RoleToken should be created from the passed
+	// RoleName. Nomad will renew this token at half its lease lifetime.
+	RoleToken string `mapstructure:"role_token"`
+
+	// AllowUnauthenticated allows users to submit jobs requiring Vault tokens
+	// without providing a Vault token proving they have access to these
+	// policies.
+	AllowUnauthenticated bool `mapstructure:"allow_unauthenticated"`
+
+	// ChildTokenTTL is the TTL of the tokens created by Nomad Servers and used
+	// by the client.  There should be a minimum time value such that the client
+	// does not have to renew with Vault at a very high frequency
+	ChildTokenTTL string `mapstructure:"child_token_ttl"`
+
+	// Addr is the address of the local Vault agent
+	Addr string `mapstructure:"address"`
+
+	// CACert is the path to a PEM-encoded CA cert file to use to verify the
+	// Vault server SSL certificate.
+	CACert string `mapstructure:"ca_cert"`
+
+	// CAPath is the path to a directory of PEM-encoded CA cert files to verify
+	// the Vault server SSL certificate.
+	CAPath string `mapstructure:"ca_path"`
+
+	// CertFile is the path to the certificate for Vault communication
+	CertFile string `mapstructure:"cert_file"`
+
+	// KeyFile is the path to the private key for Vault communication
+	KeyFile string `mapstructure:"key_file"`
+
+	// VerifySSL enables or disables SSL verification
+	VerifySSL bool `mapstructure:"verify_ssl"`
```

Comment:
Vault uses `tls_skip_verify` for this.
Can we rename this and do do `Insecure:      c.TLSSkipVerify` instead of `Insecure:      !c.VerifySSL`?